pages/reset-password.js

```diff
@@ -99,17 +99,9 @@ class ResetPasswordPage extends React.Component {
                   <Image src="/static/images/oc-logo-watercolor-256.png" height={128} width={128} />
                 </Flex>
 
-                <Flex
-                  as="label"
-                  fontWeight={700}
-                  htmlFor="password"
-                  fontSize={'32px'}
-                  mb={12}
-                  mt={3}
-                  justifyContent="center"
-                >
+                <H1 fontWeight={700} fontSize="32px" mb={12} mt={3} textAlign="center">
```

Comment:
The signin page is also constructed like this. With `as="label"`

Reply:
Semantically it is not ideal: the page has no h1, and there's already a label for the input. (from https://github.com/opencollective/opencollective-frontend/blob/70b4db981d31cfda5e0757fbc4ac4fdeb2f81f3e/pages/reset-password.js#L164)

Reply:
I made the change as it also made it easier to build the selector, but can move it to a different PR if it seems out of scppe

Reply:
I don't mind, just wanted to highlight that this pattern was also used there https://github.com/opencollective/opencollective-frontend/blob/main/components/SignIn.js#L147

So maybe we ultimately want to change that one too.

Reply:
This one is more tricky to update because the `SignIn` component could be used in different places that already define a `h1`